tools/data-model-exporter/data_model_exporter/tests/test_property.py

```diff
@@ -0,0 +1,121 @@
+import unittest
+
+from data_model_exporter.property import Property
+from data_model_exporter.property_types import PrimitiveType, RefType
+
+
+class PropertyTestCase(unittest.TestCase):
+    def test_is_array_type_true_if_either_cardinality(self):
+        self.assertTrue(Property("foo", "desc", min_cardinality=1).is_array_type())
+        self.assertTrue(Property("foo", "desc", max_cardinality=3).is_array_type())
+
+    def test_is_array_type_false_if_neither_cardinality(self):
+        self.assertFalse(Property("foo", "desc").is_array_type())
+
+    def test_add_type_adds_type(self):
+        prop = Property("foo", "desc")
+        self.assertEqual(len(prop.allowed_types), 0)
+        prop.add_type(PrimitiveType("string"))
+        self.assertEqual(prop.allowed_types, [PrimitiveType("string")])
+
+    def test_add_type_blocks_further_types_if_restrictive(self):
+        prop = Property("foo", "desc")
+        prop.add_type(PrimitiveType("string"), restrictive=True)
+        with self.assertRaises(ValueError):
+            prop.add_type(RefType("argbl"))
+
+    def test_add_type_allows_further_types_if_not_restrictive(self):
+        prop = Property("foo", "desc")
+        prop.add_type(PrimitiveType("string"), restrictive=False)
+        # we just run this to see if it raises an exception, so there's no assertions in the test
+        prop.add_type(RefType("argbl"))
```

Comment:
on further consideration, we should probably add an assertion here to make sure that the second type was actually added to the allowed types list